### PR TITLE
Use new dind img for jellyfish owned gc vm pipelines

### PR DESCRIPTION
**Pull request by lindnerby**
**Description**

Changes proposed in this pull request:
- Replace execution of Jellyfish owned pipelines running on GCloud VMs with new Docker-in-Docker images
    - unstable-cli
    - pre-main-reconciler-integration-k3d
    - pre-main-reconciler-cli-integration-kyma-latest-k3d
    - pre-main-reconciler-cli-integration-kyma-main-k3d
    - pre-main-reconciler-cli-kyma-prev-to-last-release-upgrade-k3d
    - kyma-upgrade-k3d-kyma2-to-main
- Delete unused scripts and template configs

**Related issue(s)**
* https://github.com/kyma-project/lifecycle-manager/issues/682

Scripts were added to the related repos:
- https://github.com/kyma-incubator/reconciler/pull/1372
- https://github.com/kyma-project/kyma/pull/17825
- https://github.com/kyma-project/cli/pull/1708

### CODE REVIEW

File: prow/jobs/incubator/reconciler/reconciler-k3d.yaml [hunk 1]
```diff
@@ -0,0 +1,297 @@
...
+                pushd "$install_dir" || exit
+                curl -Lo kyma "${kyma_cli_url}"
+                chmod +x kyma
+                popd || exit
+                kyma version --client
+                kyma provision k3d --ci
+                kyma deploy --ci --concurrency=8 --profile=evaluation --source="${KYMA_SOURCE}"
+                pushd "/home/prow/go/src/github.com/kyma-project/kyma/tests/fast-integration"
+                git remote add origin https://github.com/kyma-project/kyma.git
+                git reset --hard && git remote update && git fetch --all >/dev/null 2>&1 && git checkout "${KYMA_SOURCE}"
```
Comment (Ressetkk) [inline prow/jobs/incubator/reconciler/reconciler-k3d.yaml:129]: Im thinking if isn't it just better to define the kyma version explicitly in the base_ref of extra_refs. Then this part would simply go down to:
```
kyma deploy --ci --profile=evaluation --source=local
make -C tests/fast-integration ci
```

File: prow/jobs/incubator/reconciler/reconciler-k3d.yaml [hunk 2]
```diff
@@ -0,0 +1,297 @@
...
+                make ci
+                popd
+            env:
+              - name: GO11MODULE
+                value: "on"
```
Comment (Ressetkk) [inline prow/jobs/incubator/reconciler/reconciler-k3d.yaml:134]: this variable is not needed anymore

**Review by Ressetkk — CHANGES_REQUESTED** [review]
I haven't looked at the rest, as I would be generally repeating myself in the comments at this point. Please check all newly added ProwJobs with my suggestions that I published for several of them.

I see a lot of inconsistencies between these scripts like usage of `make` for running fast integration test either by `cd`'ing into the tests/fast-integration dir or using `-C` parameter. Please use either one or another, so you are consistent. Also I see that they are quite big. Maybe it's a good idea to move them away to external file once they are working.

File: prow/jobs/incubator/reconciler/reconciler-k3d.yaml (hunk 1 above)
Comment (nesmabadr) [reply prow/jobs/incubator/reconciler/reconciler-k3d.yaml]: @Ressetkk But the kyma version here is fetched from the releases API (to get the latest release), how can I add the base_ref as the latest release in this case?

File: prow/jobs/cli/cli-k3d.yaml [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+              privileged: true
+              seccompProfile:
+                type: Unconfined
+              allowPrivilegeEscalation: true
+            args:
+              - "bash"
+              - "-c"
+              - |
+                chmod +x ./prow/publish-unstable-cli.sh
+                ./prow/publish-unstable-cli.sh
```
Comment (Ressetkk) [inline prow/jobs/cli/cli-k3d.yaml:51]: Chmod your script locally and push it to target repo. Then you don't need to chmod it in here.
```suggestion
              - ./prow/publish-unstable-cli.sh
```

File: prow/jobs/cli/cli-k3d.yaml [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+        - ^master$
+        - ^main$
+      extra_refs:
+        - org: kyma-project
+          repo: kyma
+          path_alias: github.com/kyma-project/kyma
+          base_ref: main
+        - org: kyma-incubator
+          repo: reconciler
+          path_alias: github.com/kyma-incubator/reconciler
```
Comment (Ressetkk) [inline prow/jobs/cli/cli-k3d.yaml:32]: This is not needed

File: prow/jobs/cli/cli-k3d.yaml [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+      decorate: true
+      cluster: trusted-workload
+      max_concurrency: 10
+      branches:
+        - ^master$
+        - ^main$
+      extra_refs:
+        - org: kyma-project
+          repo: kyma
+          path_alias: github.com/kyma-project/kyma
```
Comment (Ressetkk) [inline prow/jobs/cli/cli-k3d.yaml:28]: This is not needed

File: prow/jobs/incubator/reconciler/reconciler-k3d.yaml [hunk 6]
```diff
@@ -0,0 +1,182 @@
...
+              privileged: true
+              seccompProfile:
+                type: Unconfined
+              allowPrivilegeEscalation: true
+            args:
+              - "bash"
+              - "-c"
+              - |
+                chmod +x ./scripts/prow/cli-integration-latest.sh
+                ./scripts/prow/cli-integration-latest.sh
```
Comment (Ressetkk) [inline prow/jobs/incubator/reconciler/reconciler-k3d.yaml:87]: Chmod your script locally and push it to target repo. Then you don't need to chmod it in here.
```suggestion
              - ./scripts/prow/cli-integration-latest.sh
```

File: prow/jobs/incubator/reconciler/reconciler-k3d.yaml [hunk 7]
```diff
@@ -0,0 +1,182 @@
...
+      skip_report: false
+      decorate: true
+      cluster: untrusted-workload
+      max_concurrency: 10
+      branches:
+        - ^main$
+      extra_refs:
+        - org: kyma-project
+          repo: kyma
+          path_alias: github.com/kyma-project/kyma
```
Comment (Ressetkk) [inline prow/jobs/incubator/reconciler/reconciler-k3d.yaml:71]: Not needed

File: prow/jobs/incubator/reconciler/reconciler-k3d.yaml [hunk 8]
```diff
@@ -0,0 +1,182 @@
...
```
Comment (Ressetkk) [inline prow/jobs/incubator/reconciler/reconciler-k3d.yaml:118]: Not needed

File: prow/jobs/incubator/reconciler/reconciler-k3d.yaml [hunk 9]
```diff
@@ -0,0 +1,182 @@
...
+              privileged: true
+              seccompProfile:
+                type: Unconfined
+              allowPrivilegeEscalation: true
+            args:
+              - "bash"
+              - "-c"
+              - |
+                chmod +x ./scripts/prow/cli-integration-release-upgrade.sh
+                ./scripts/prow/cli-integration-release-upgrade.sh
```
Comment (Ressetkk) [inline prow/jobs/incubator/reconciler/reconciler-k3d.yaml:134]: Chmod your script locally and push it to target repo. Then you don't need to chmod it in here.
```suggestion
              - ./scripts/prow/cli-integration-release-upgrade.sh
```

File: prow/jobs/incubator/reconciler/reconciler-k3d.yaml [hunk 10]
```diff
@@ -0,0 +1,182 @@
...
+              privileged: true
+              seccompProfile:
+                type: Unconfined
+              allowPrivilegeEscalation: true
+            args:
+              - "bash"
+              - "-c"
+              - |
+                chmod +x ./scripts/prow/reconciler-integration.sh
+                ./scripts/prow/reconciler-integration.sh
```
Comment (Ressetkk) [inline prow/jobs/incubator/reconciler/reconciler-k3d.yaml:175]: Chmod your script locally and push it to target repo. Then you don't need to chmod it in here.
```suggestion
              - ./scripts/prow/reconciler-integration.sh
```

File: prow/jobs/kyma/kyma-integration-k3d.yaml [hunk 11]
```diff
@@ -0,0 +1,64 @@
...
+            privileged: true
+            seccompProfile:
+              type: Unconfined
+            allowPrivilegeEscalation: true
+          args:
+            - "bash"
+            - "-c"
+            - |
+              chmod +x ./tests/fast-integration/prow/kyma-upgrade.sh
+              ./tests/fast-integration/prow/kyma-upgrade.sh
```
Comment (Ressetkk) [inline prow/jobs/kyma/kyma-integration-k3d.yaml:54]: Chmod your script locally and push it to target repo. Then you don't need to chmod it in here.
```suggestion
            - ./tests/fast-integration/prow/kyma-upgrade.sh
```

File: prow/jobs/kyma/kyma-integration-k3d.yaml [hunk 12]
```diff
@@ -0,0 +1,64 @@
...
+    decorate: true
+    decoration_config:
+      grace_period: 600000000000
+      timeout: 7200000000000
+    path_alias: github.com/kyma-project/kyma
+    cluster: trusted-workload
+    extra_refs:
+      - org: kyma-project
+        repo: kyma
+        path_alias: github.com/kyma-project/kyma
```
Comment (Ressetkk) [inline prow/jobs/kyma/kyma-integration-k3d.yaml:31]: Not needed

File: prow/jobs/kyma/kyma-integration-k3d.yaml [hunk 13]
```diff
@@ -0,0 +1,64 @@
...
+        base_ref: main
+      - org: kyma-incubator
+        repo: local-kyma
+        path_alias: github.com/kyma-incubator/local-kyma
```
Comment (Ressetkk) [inline prow/jobs/kyma/kyma-integration-k3d.yaml:35]: not needed

**Review by Ressetkk — CHANGES_REQUESTED** [review]

File: prow/jobs/kyma/kyma-integration-k3d.yaml (hunk 12 above)
Comment (nesmabadr) [reply prow/jobs/kyma/kyma-integration-k3d.yaml:31]: that's where the script is located, so it needs the kyma repo.

File: prow/jobs/incubator/reconciler/reconciler-k3d.yaml (hunk 8 above)
Comment (nesmabadr) [reply prow/jobs/incubator/reconciler/reconciler-k3d.yaml:110]: needed to run the fast integration tests

**Review by Ressetkk — APPROVED** [review]

**Pull request merged**